File: c++/Solution167.cpp
//LR Pointer O(n) O(1)
class Solution167 {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int sum = nums[l] + nums[r];
            if (sum == target) {
                return {l + 1, r + 1};
            } else if (sum < target) {
                ++l;
            } else { // sum > target
                --r;
            }
        }
        return {-1, -1};
    }
};

File: java/Solution167.java
package leetCode.java;

//LR Pointer O(n) O(1)
class Solution167 {
    public int[] twoSum(int[] nums, int target) {
        // Input Array Is Sorted, 其實就是3sum的內層算法
        int l = 0, r = nums.length - 1;
        while (l < r) {
            if (nums[l] + nums[r] == target) {
                return new int[]{l + 1, r + 1};
            }else if (nums[l] + nums[r] < target) {
                ++l;
            } else {
                --r;
            }
        }

        return new int[]{};
    }
}